tests/kernel/mem_protect/syscalls/src/main.c

```diff
@@ -112,6 +112,34 @@ static inline int z_vrfy_syscall_arg64(u64_t arg)
 }
 #include <syscalls/syscall_arg64_mrsh.c>
 
+ /* System call made for test test_system_call_separate_stack */
+int z_impl_syscall_separate_stack(void)
+{
+	/* Test that syscall is running on a separate stack, not accessible
+	 * to the calling thread.
+	 */
+	int syscall_var = 10;
+	int *var_ptr = &syscall_var;
+	intptr_t ptr_int = (intptr_t)var_ptr;
+
+	struct k_thread *thread = _current;
+	unsigned int start_addr = thread->stack_info.start;
+	size_t size_addr = thread->stack_info.size;
+
+	if ((ptr_int >= (start_addr + size_addr)) || (ptr_int <= start_addr)) {
```

Comment:
I think this might be too loose, maybe check explicitly that the variable is inside the privileged stack?

Reply:
Could you send me your test name and file where it is placed in codebase? I didn't know about it.

Reply:
How to access privileged stack?

Reply:
he wants you to validate that the stack pointer is within the bounds of the privilege stack.

we currently do not have a generic function for obtaining the privilege stack bounds. in some cases you call z_priv_stack_find() and others you do math on the base address of thread->stack_obj. it varies per architecture. can you file a bug on this and assign to me?

Reply:
@andrewboie we have arch.priv_stack_start for ARC, and ARM (not x86 AFAICS). So this could be used in this test, to add more value to the check.

Reply:
@andrewboie @ioannisg I made changes for ARM and ARC, please review my new commit. As I understand there is no option for x86, and I will file an issue and assign it to Andrew to develop arch.priv_stack_start for x86.